go_util/util.go

```diff
@@ -31,3 +36,167 @@ func AppendPendingMetadata(testName string, link shared.MetadataLink, metadata *
+	deletedMetadataLinks := shared.MetadataLinks{}
+	newMetadataLinks := []shared.MetadataLink{}
+	for _, link := range metadata.Links {
+		deletedResults := []shared.MetadataTestResult{}
+		newResults := []shared.MetadataTestResult{}
+		for _, result := range link.Results {
+			if result.TestPath == testname {
+				// Add the deleted result to the metadataLinks
+				deletedResults = append(deletedResults, result)
+			} else {
+				newResults = append(newResults, result)
+			}
+		}
+		if len(deletedResults) > 0 {
+			deletedLink := shared.MetadataLink{Product: link.Product, URL: link.URL, Results: deletedResults}
+			deletedMetadataLinks = append(deletedMetadataLinks, deletedLink)
+		}
+
+		if len(newResults) > 0 {
+			newLink := shared.MetadataLink{Product: link.Product, URL: link.URL, Results: newResults}
+			newMetadataLinks = append(newMetadataLinks, newLink)
+		}
+	}
+	newMetadata := shared.Metadata{Links: newMetadataLinks}
+	return newMetadata, deletedMetadataLinks
+}
+
+// DeleteTestFromMetadata deletes a WPT testPath and returns the deleted metadata entries, if exists.
+func DeleteTestFromMetadata(testPath string) shared.MetadataLinks {
+	metadata := readMetadataFile(testPath)
+	if metadata == nil {
+		return shared.MetadataLinks{}
+	}
+
+	testname := path.Base(testPath)
+	newMetadata, metadataLinks := deleteMetadata(testname, *metadata)
+	// Nothing is deleted.
+	if len(metadataLinks) == 0 {
+		return shared.MetadataLinks{}
+	}
+
+	fmt.Println("Removing", testPath, "from the wpt-metadata repo")
+	// Everything is deleted
+	if len(newMetadata.Links) == 0 {
+		removeMetadataFile(testPath)
+	} else {
+		filepath := getYMLFilePath(testPath)
+		f, err := os.OpenFile(filepath, os.O_WRONLY|os.O_TRUNC, 0644)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer f.Close()
+		writeToFile(newMetadata, f)
+	}
+	return metadataLinks
+}
+
+// writeToFile writes out a wpt.fyi shared.Metadata entry to a given file handle.
+//
+// TODO(kyleju): https://github.com/web-platform-tests/wpt.fyi/issues/1957.
+func writeToFile(metadata shared.Metadata, f *os.File) {
+	metadataBytes, err := yaml.Marshal(metadata)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	_, err = f.WriteAt(metadataBytes, 0)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func readMetadataFile(testPath string) *shared.Metadata {
+	filepath := getYMLFilePath(testPath)
+	if _, err := os.Stat(filepath); os.IsNotExist(err) {
+		// The file does not exist.
+		return nil
+	}
+
+	f, err := os.OpenFile(filepath, os.O_RDONLY, 0755)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+
+	data, err := ioutil.ReadAll(f)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var metadata shared.Metadata
+	err = yaml.Unmarshal(data, &metadata)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return &metadata
+}
+
+func removeMetadataFile(testPath string) {
+	filepath := getYMLFilePath(testPath)
+	fmt.Println("Delete", filepath, "from the wpt-metadata repo")
```

Comment:
```suggestion
	fmt.Println("Removing", filepath, "from the wpt-metadata repo")
```

To match the previous similar comment.